### ISSUE TITLE
unable to connect to database

### ISSUE THREAD

**Issue by scns** [jehy/hass-addons#4]
After installation i recieve the following error:

Database is running on an external host. (MariaDB)

My recorder.yaml starts with:

db_url: mysql://USER:SECRET@192.168.99.57:3307/homeassistant?charset=utf8

___________________
-----------------------------------------------------------
 Add-on: DbStats
 Database stats for Home Assistant
-----------------------------------------------------------
 Add-on version: 0.4.5
 You are running the latest version of this add-on.
 System: Home Assistant OS 12.0  (amd64 / generic-x86-64)
 Home Assistant Core: 2024.2.5
 Home Assistant Supervisor: 2024.02.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nginx.sh
ingress_entry api/hassio_ingress/fNVZ_WEgRm-bUezPoYKEedLYw3c5alDrFmemENmxIkM
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun nginx (no readiness notification)
services-up: info: copying legacy longrun nodejs (no readiness notification)
[21:02:29] INFO: Starting NGinx...
[21:02:29] INFO: Starting dbstats...
s6-rc: info: service legacy-services successfully started
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:62
2024/02/28 21:02:29 [notice] 175#175: using the "epoll" event method
2024/02/28 21:02:29 [notice] 175#175: nginx/1.24.0
2024/02/28 21:02:29 [notice] 175#175: OS: Linux 6.6.16-haos
2024/02/28 21:02:29 [notice] 175#175: getrlimit(RLIMIT_NOFILE): 1073741816:1073741816
2024/02/28 21:02:29 [notice] 175#175: start worker processes
2024/02/28 21:02:29 [notice] 175#175: start worker process 199
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] database options from env not found, checking config
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] config file found, parsing
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/configuration.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/configuration.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/automations.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/automations.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/scripts.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/scripts.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/scenes.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/scenes.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/sensor.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/sensor.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/recorder.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/recorder.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] Input: abspath /homeassistant/templates.yaml homeDir /homeassistant
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] found and parsed secrets file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] 7 secrets found for file /homeassistant/templates.yaml
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] parsed config file
[Nest] 177  - 02/28/2024, 9:02:31 PM     LOG [ConfigService] recorder options not found, gonna try standard database path
/usr/src/app/dist/backend/src/config.service.js:231
            throw new Error(`SQLite database file not found in ${sqliteFileName}`);
            ^

Error: SQLite database file not found in /homeassistant/home-assistant_v2.db
    at ConfigService.extractRecorderConfig (/usr/src/app/dist/backend/src/config.service.js:231:19)
    at new ConfigService (/usr/src/app/dist/backend/src/config.service.js:39:36)
    at Object.<anonymous> (/usr/src/app/dist/backend/src/config.js:4:16)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/usr/src/app/dist/backend/src/app.module.js:14:18)

Node.js v18.19.0
[21:02:31] WARNING: dbstats crashed, halting add-on
[21:02:31] INFO: dbstats stopped, restarting...
s6-rc: info: service legacy-services: stopping
2024/02/28 21:02:31 [notice] 175#175: signal 15 (SIGTERM) received from 172, exiting
2024/02/28 21:02:31 [notice] 199#199: exiting
2024/02/28 21:02:31 [notice] 199#199: exit
2024/02/28 21:02:31 [notice] 175#175: signal 17 (SIGCHLD) received from 199
2024/02/28 21:02:31 [notice] 175#175: worker process 199 exited with code 0
2024/02/28 21:02:31 [notice] 175#175: exit
if: fatal: unable to spawn s6-test: No such file or directory
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Comment by jehy** [comment]
Hi! How do you include your recorder.yaml? There are too many ways to do it and I was not able to support all of them.
If you can attach recorder.yaml and configuration.yaml without sensitive data, I can try to debug it.

Meanwhile, you can try passing database URL directly on addon options page, it should work - then addon will use provided value and not try to guess database options from configs.

**Comment by scns** [comment]
This is the start of the configuration.yaml

configuration.yaml:

```yaml
# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate

#notify: !include notifications.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml
sensor: !include sensor.yaml
recorder: !include recorder.yaml
#vacuum: !include vacuum.yaml
template: !include templates.yaml

utility_meter:


```

**Comment by scns** [comment]
and the start of the recorder.yaml

```yaml
db_url: mysql://root:PASSWORD@192.168.99.57:3307/homeassistant?charset=utf8
purge_keep_days: 14
commit_interval: 30
include:
  entities:
    - sensor.ospel_wind_direction
    - sensor.voltage_phase_l1
```

**Comment by jehy** [comment]
Hi! Thanks for the config, I was able to reproduce this issue and fixed it in version 0.4.6 - please try it :)

**Comment by scns** [comment]
Yes! It works now! Tnx!
